apps/app/src/components/Header/UserMenu.tsx

```diff
@@ -0,0 +1,71 @@
+import { Transition } from "@headlessui/react";
+import { useEffect, useRef, useState } from "react";
+import FocusLock from "react-focus-lock";
+import { UserData } from "../../types";
+import { useOnClickOutside } from "../../hooks/useOnClickOutside";
+import { useUser } from "../../hooks/useUser";
+import { UserAvatar } from "./UserAvatar";
+
+type UserMenuProps = {
+	userData: UserData;
+};
+
+export const UserMenu = ({ userData }: UserMenuProps) => {
+	const { logout } = useUser();
+	const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+	const dropdownRef = useRef<HTMLDivElement>(null);
+	useOnClickOutside(dropdownRef, () => isDropdownVisible && setIsDropdownVisible(false));
+
+	useEffect(() => {
+		const handleEscapeKeyPress = ({ key }: KeyboardEvent) =>
+			isDropdownVisible && key === "Escape" && setIsDropdownVisible(false);
+
+		window.addEventListener("keydown", handleEscapeKeyPress);
+
+		return () => window.removeEventListener("keydown", handleEscapeKeyPress);
+	}, [isDropdownVisible]);
+
+	return (
+		<div ref={dropdownRef} className="flex flex-col items-center gap-0">
+			<button
+				type="button"
+				className="relative mx-auto flex flex flex-row items-center gap-3 rounded-full bg-violet-800 pr-6 uppercase transition-opacity hover:opacity-80 dark:bg-violet-700 sm:gap-2 sm:pr-4"
+				onClick={() => setIsDropdownVisible((prev) => !prev)}
+				aria-label={`${isDropdownVisible ? "Zamknij" : "Otwórz"} menu użytkownika`}
+				aria-expanded={isDropdownVisible}
+				aria-controls="user-menu"
+			>
+				<div className="relative h-12 w-12 sm:h-6 sm:w-6">
```

Comment:
Trochę małe zdjęcie na desktopie

Reply:
Przy dobieraniu rozmiaru zdjęcia sugerowałem się tym, aby zachować symetrie w tym menu, ale zgadzam się, że nie wygląda to może do końca tak, jak powinno. Nie miałem innego pomysłu.

![Group 1](https://user-images.githubusercontent.com/27455716/211895980-220d34b6-6355-4130-ac00-477872c9954d.png)

Zastanawia mnie również to, czy to menu kontekstowe nie jest trochę niepotrzebne, bo kolejne opcje dodawane sią do "menu" w komponencie `CtaHeader`. Moim zdaniem można by to jakoś ujednolicić. Jeśli w `CtaHeader` znajdzie się więcej opcji to będzie trzeba pomyśleć o jego przebudowie szczególnie na smartfony. Możnaby ewentualnie część opcji związanych z użytkownikiem przerzucić do menu kontekstowego 🤔